cdap-master/src/main/java/co/cask/cdap/data/tools/QueueConfigUpgrader.java

```diff
@@ -61,7 +61,13 @@ protected TableId getTableId() {
 
   @Nullable
   private QueueName fromRowKey(byte[] oldRowKey) {
-    QueueName queueName = QueueName.from(oldRowKey);
+    QueueName queueName;
+    try {
+      queueName = QueueName.from(oldRowKey);
```

Comment:
The rewrite of the data happens before or after the table get upgrade to dataset table (i.e. before or after QueueAdmin.upgrade() is called)? If it is before, then it should be sufficient to just check the version.